default/AI/AIFleetMission.py

```diff
@@ -474,22 +474,34 @@ def generate_fleet_orders(self):
                 fleet_order = self._get_fleet_order_from_target(self.type, self.target)
                 self.orders.append(fleet_order)
 
-    def _need_repair(self):
-        """
-        Check if fleet need repair. Check if fleet HP is less of cutoff.
-        TODO make more clever implementation.
-        """
-        repair_limit = 0.70
+    def _need_repair(self, repair_limit=0.70):
+        """Check if fleet needs to be repaired.
+
+         If the fleet is already at a system where it can be repaired, stay there until fully repaired.
+         Otherwise, repair if fleet health is below specified *repair_limit*.
+         For military fleets, there is a special evaluation called, cf. *MilitaryAI.avail_mil_needing_repair()*
 
+         :param repair_limit: percentage of health below which the fleet is sent to repair
+         :type repair_limit: float
+         :return: True if fleet needs repair
+         :rtype: bool
+        """
+        # TODO: More complex evaluation if fleet needs repair (consider self-repair, distance, threat, mission...)
         universe = fo.getUniverse()
         fleet_id = self.fleet.id
+        # if we are already at a system where we can repair, make sure we use it...
+        system = self.fleet.get_system()
+        nearest_dock = MoveUtilsAI.get_nearest_drydock_system_id(system.id)
```

Comment:
may be add `TODO: obstruction may be ignored here if fleet not in drydocked system` and add comment about obstruction to docstring of `get_nearest_drydock_system_id`